### ISSUE TITLE
S7KReader is dependent on 7200 record

### ISSUE THREAD

**Issue by buurgz** [Teledyne-Marine/pyread7k#35]
file_catalog method of S7KReader assumes there will be a file_header (7200 record) at position 0 in the file. Tested this using a Norbit generated file where the first record in the file is 7000.

**Comment by rasmus-klett-tele** [comment]
That is interesting, I've never seen a file without a 7200 record. Would you be able to share it?

Have you tried the `pyread7k.S7KRecordReader`? It is a simple linear reader which does not assume the presence of 7200 like the FileDataset does. It of course does not have the nice automatic grouping of data into pings like the FileDataset. It might be possible to support 7200-less files in the dataset classes as well.

**Comment by buurgz** [comment]
Yes I did find `pyread7k.S7KRecordReader ` thanks. I thought I might be able to make some changes to make it work without the 7200 record but there's a few libraries in use that I'm unfamiliar with so I got stuck. I've attached an example file
[20200224_165112.zip](https://github.com/Teledyne-Marine/pyread7k/files/8602158/20200224_165112.zip)
.

**Comment by rasmus-klett-tele** [comment]
I can give you some pointers if you want to make a PR yourself? That would be the fastest way for you, I don't have time for it this week.

I think it's actually a pretty simple change:

1. In `S7KReader` make the type of `file_header` Optional. 
2. Return None if the read of record 7200 fails.
3. In `S7KReader.file_catalog` if the header is None, call `self._build_file_catalog` just like if the `try` block throws an exception.

This will cause the FileDataset to read through the whole dataset and build a file catalog for random-access. Depending on your use case, it might be a bit slow, but it's very easy to implement, and gives you the full power of the FileDataset abstraction.

I'll be very interested to hear if you run into other problems with Norbit data, I haven't worked with it at all.

**Comment by buurgz** [comment]
Yes, that was simple! Thanks for your guidance with that. I've made a pull request; the first time for me on someone else's project so I hope I did it correctly. Having everything associated by ping number is much easier!

**Comment by rasmus-klett-tele** [comment]
Thanks a lot for your contribution, you did it correctly :) I [changed it slightly](https://github.com/Teledyne-Marine/pyread7k/commit/7668eace85e6b8dce794c302900e5e7150f10a8d) before merging, to make it warn/raise by default when 7200 is missing. That means you will have to add an argument to the dataset if you want to allow missing 7200:

```python
dataset = FileDataset(
    "my_path.s7k",
    catalog_issue_handling=pyread7k.CatalogIssueHandling.HANDLE_CORRUPT,
)
```

I know it's slightly inconvenient for you, but since the handling is a lot slower, it is important that the user has actively chosen it. For example, if you are running this code in a cloud environment you could end up paying extra without realizing because your files dont have 7200/7300 records.

If you have files that you are going to read many times (I have a lot of those use cases personally), I recommend pickling the dataset after instantiation, so that next time you can just unpickle and don't have to reconstruct the file catalog again.

**Comment by rasmus-klett-tele** [comment]
For the record, the change was merged in PR #36
